### PR TITLE
Add conda-forge CI test

### PR DESCRIPTION
**Pull request by johnwason**
This PR adds testing for conda-forge builds, and patches issues that were found during conda builds.

### CODE REVIEW

File: tesseract_geometry/cmake/Findassimp.cmake [hunk 1]
```diff
@@ -1,3 +1,7 @@
+find_package(assimp QUIET CONFIG)
+
+if(NOT assimp_FOUND)
```
Comment (Levi-Armstrong) [inline tesseract_geometry/cmake/Findassimp.cmake:3]: Should this also check if the target `assimp::assimp` exists and if not create even if found?

File: tesseract_geometry/include/tesseract_geometry/mesh_parser.h [hunk 2]
```diff
@@ -157,10 +157,10 @@ std::vector<std::shared_ptr<T>> extractMeshData(const aiScene* scene,
     {
       aiMaterial* mat = scene->mMaterials[a->mMaterialIndex];
       {
-        Eigen::Vector4d base_color;
+        Eigen::Vector4d base_color = Eigen::Vector4d::Zero();
         double metallic = 0.0;
         double roughness = 0.5;
-        Eigen::Vector4d emissive;
+        Eigen::Vector4d emissive = Eigen::Vector4d::Zero();
```
Comment (Levi-Armstrong) [inline tesseract_geometry/include/tesseract_geometry/mesh_parser.h:163]: I think these both need to be initialized to make clang-tidy happy.

**Review by Levi-Armstrong — CHANGES_REQUESTED** [review]
Also cmake format needs to be ran.

File: tesseract_geometry/cmake/Findassimp.cmake (hunk 1 above)
Comment (johnwason) [reply tesseract_geometry/cmake/Findassimp.cmake:3]: Good idea, I will add the check

File: tesseract_geometry/include/tesseract_geometry/mesh_parser.h (hunk 2 above)
Comment (johnwason) [reply tesseract_geometry/include/tesseract_geometry/mesh_parser.h:163]: I was getting failures with the GCC compiler in conda. I guess it has stricter checks.

**Comment by johnwason** [comment]
@Levi-Armstrong more tests failed than I was expecting, I will take a look this week or next.

**Comment by johnwason** [comment]
I think this should be ready now

**Pull request merged**